hw_diag/utilities/hardware.py

```diff
@@ -373,6 +380,97 @@ def is_nebra_device(diagnostics: dict) -> bool:
     return "nebra" in friendly_name.lower()
 
 
+def has_external_antenna_support() -> bool:
+    device_type = os.getenv('BALENA_DEVICE_TYPE')
+    return device_type in EXT_ANT_DEVICE_TYPES
+
+
+def is_external_antenna_enabled() -> bool:
+    bc = balena_cloud.BalenaCloud.new_from_env()
+    variables = bc.get_device_config_variables()
+    for var_dict in variables:
+        if var_dict['name'] not in DTPARAM_CONFIG_VAR_NAMES:
+            continue
+
+        # dtparam var value is a comma-separated list of values to be used as separate
+        # `dtparam=<val>` lines in `config.txt`
+        return EXT_ANT_DTPARAM in var_dict['value']
+
+    return False
+
+
+def _get_device_dtparams_var(bc: balena_cloud.BalenaCloud) -> tuple[list[str], int]:
+    # BALENA_HOST_CONFIG_dtparam/RESIN_HOST_CONFIG_dtparam var value is a comma-separated list of
+    # values to be used as separate `dtparam=<val>` lines in `config.txt`.
+    variables = bc.get_device_config_variables()
+    for var_dict in variables:
+        if var_dict['name'] not in DTPARAM_CONFIG_VAR_NAMES:
+            continue
+
+        device_dtparams_var_id = var_dict['id']
+
+        value = var_dict['value'].strip()
+        if value:
+            device_dtparams = value.split(',')
+        else:
+            device_dtparams = []
+        break
+    else:
+        device_dtparams = []
+        device_dtparams_var_id = None
+
+    return device_dtparams, device_dtparams_var_id
+
+
+def _get_fleet_dtparams_var(bc: balena_cloud.BalenaCloud) -> list[str]:
+    variables = bc.get_fleet_config_variables()
+    for var_dict in variables:
+        if var_dict['name'] not in DTPARAM_CONFIG_VAR_NAMES:
+            continue
+
+        value = var_dict['value'].strip()
+        if value:
+            return value.split(',')
+        else:
+            return []
+
+    return []
+
+
+def set_external_antenna_enabled(enabled: bool) -> None:
+    bc = balena_cloud.BalenaCloud.new_from_env()
+
+    # The device variable overrides the fleet variable, so we need to deal with both.
+    device_dtparams, device_dtparams_var_id = _get_device_dtparams_var(bc)
+    fleet_dtparams = _get_fleet_dtparams_var(bc)
+
+    if enabled:
+        logging.info('Enabling external antenna')
+        if EXT_ANT_DTPARAM in device_dtparams:
+            logging.info('External antenna already enabled')
+            return
+
+        # If we've already got device dtparams that override fleet ones, simply append ours;
+        # if there's no device override yet, we create the override by copying the fleet dtparams.
+        if device_dtparams:
+            device_dtparams.append(EXT_ANT_DTPARAM)
+        else:
+            device_dtparams = fleet_dtparams + [EXT_ANT_DTPARAM]
+    else:
+        logging.info('Disabling external antenna')
+        try:
+            device_dtparams.remove(EXT_ANT_DTPARAM)
```

Comment:
this is a little strange. But there is a chance that device is left with parameters that are not in sycn with fleet even when there are no potential antenna overrides.

Reply:
@pritamghanghas do you mean that we should remove all overrides and just revert to the fleet settings once the external antenna parameter is removed?

If so, that's probably a good idea actually

Reply:
Indeed that's a very good idea and that's the first thing that I wanted to do here. However, the `DELETE` API call returns 403 (yes, I can create and update vars but I can't delete them). Hence the workaround.

Reply:
How strange. 

Guess that's just another annoying quirk of Balena!

If no other way, then this seems like the only one TBH